docs/articles_en/documentation/openvino-ir-format/operation-sets/operations-specifications/pooling/max-pool-14.rst

```diff
@@ -0,0 +1,404 @@
+      ``H_out = ceil((H - ((kernel[0] - 1) * dilations[0] + 1) + 1) / strides[0])``
+      ``W_out = ceil((W - ((kernel[1] - 1) * dilations[1] + 1) + 1) / strides[1])``
+      ``D_out = ceil((D - ((kernel[2] - 1) * dilations[2] + 1) + 1) / strides[2])``
+
+* ``auto_pad = same_upper / same_lower``
+      ``H_out = H``
+      ``W_out = W``
+      ``D_out = D``
+
+
+If ``H + pads_begin[i] + pads_end[i] - kernel[i]`` is not divisible by ``strides[i]`` evenly, the result is rounded with respect to the ``rounding_type`` attribute.
+
+1. Example 1 shows how *MaxPool* operates with 4D input using 2D kernel and ``auto_pad = explicit``.
+
+   .. code-block:: sh
+
+      input = [[[[-1, 2, 3],
+                 [4, 5, -6],
+                 [-7, 8, 9]]]]
+      strides = [1, 1]
+      pads_begin = [1, 1]
+      pads_end = [1, 1]
+      kernel = [2, 2]
+      rounding_type = "floor"
+      auto_pad = "explicit"
+      output0 = [[[[-1, 2, 3, 3],
+                   [4, 5, 5, -6],
+                   [4, 8, 9, 9],
+                   [-7, 8, 9, 9]]]]
+      output1 = [[[[0, 1, 2, 2],
+                   [3, 4, 4, 5],
+                   [3, 7, 8, 8],
+                   [6, 7, 8, 8]]]]
+
+
+2. Example 2 shows how *MaxPool* operates with 3D input using 1D kernel and ``auto_pad = valid``.
+
+   .. code-block:: sh
+
+      input = [[[-1, 2, 3, 5, -7, 9, 1]]]
+      strides = [1]
+      kernel = [3]
+      rounding_type = "floor"
+      auto_pad = "valid"
+      output0 = [[[3, 5, 5, 9, 9]]]
+      output1 = [[[2, 3, 3, 5, 5]]]
+
+
+3. Example 3 shows how *MaxPool* operates with 4D input using 2D kernel and ``auto_pad = same_lower``.
+
+   .. code-block:: sh
+
+      input = [[[[-1, 2, 3],
+               [4, 5, -6],
+               [-7, 8, 9]]]]
+      strides = [1, 1]
+      kernel = [2, 2]
+      rounding_type = "floor"
+      auto_pad = "same_lower"
+      output0 = [[[[-1, 2, 3],
+                  [4, 5, 5]
+                  [4, 8, 9]]]]
+      output1 = [[[[0, 1, 2],
+                  [3, 4, 4]
+                  [3, 7, 8]]]]
+
+
+4. Example 4 shows how *MaxPool* operates with 4D input using 2D kernel and ``auto_pad = same_upper``.
+
+
+   .. code-block:: sh
+
+      input = [[[[-1, 2, 3],
+                 [4, 5, -6],
+                 [-7, 8, 9]],
+                [[2, -1, 5],
+                 [6, -7, 1],
+                 [8, 2, -3]]]]
+      strides = [1, 1]
+      kernel = [2, 2]
+      rounding_type = "floor"
+      auto_pad = "same_upper"
+      output0 = [[[[5, 5, 3],
+                   [8, 9, 9]
+                   [8, 9, 9]],
+                  [[6, 5, 5],
+                   [8, 2, 1],
+                   [8, 2, -3]]]]
+      output1 = [[[[4, 4, 2],
+                   [7, 8, 8]
+                   [7, 8, 8]],
+                  [[12, 11, 11],
+                   [15, 16, 14],
+                   [15, 16, 17]]]]
+
+
+5. Example 5 shows how *MaxPool* operates with 4D input using 2D kernel and ``rounding_type = ceil_torch``.
```

Comment:
Would be great to add corresponding examples to the AvgPool spec or create a common document with explanation of pooling shape inference rules, padding and rounding modes, then reuse it as a reference in both AvgPool and MaxPool spec, similar to the broadcast rules document [/operation-sets/broadcast-rules.rst](https://github.com/openvinotoolkit/openvino/blob/master/docs/articles_en/documentation/openvino-ir-format/operation-sets/broadcast-rules.rst) mentioned in several ops like  [/operation-sets/operations-specifications/arithmetic/add-1.rst](https://github.com/openvinotoolkit/openvino/blob/master/docs/articles_en/documentation/openvino-ir-format/operation-sets/operations-specifications/arithmetic/add-1.rst)).